### PR TITLE
Add benchmarking tool for the table plugin

### PR DESCRIPTION
**Pull request by danpat**
# Issue

Adds a simple tool to benchmark the `table` plugin.  Usage is similar to the `match` benchmark, except that the tool will work with any OSRM datafile (not coupled to Monaco).

## Tasklist
 - [ ] review
 - [ ] adjust for comments

### CODE REVIEW

File: src/benchmarks/table.cpp [hunk 1]
```diff
@@ -0,0 +1,96 @@
...
+    EngineConfig config;
+    config.storage_config = {argv[1]};
+    config.use_shared_memory = false;
+
+    // Routing machine with several services (such as Route, Table, Nearest, Trip, Match)
+    OSRM osrm{config};
+
+    // Shuffle the coordinate list randomly, then we'll simply use the first N coordinates
+    // to build our random table queries
+    std::random_shuffle(coordinates.begin(), coordinates.end());
```
Comment (oxidase) [inline src/benchmarks/table.cpp]: i think it must be moved to line 62 to have a random sample for every test run, at the moment it is not: page caching contains access history

File: src/benchmarks/table.cpp [hunk 2]
```diff
@@ -0,0 +1,96 @@
...
+
+    auto runtest = [&](const int coordcount) -> void {
+        TableParameters params;
+        for (int i = 0; i < coordcount; i++)
+        {
+            params.coordinates.push_back(coordinates[i]);
+        }
+
+        TIMER_START(routes);
+        auto NUM = 10;
```
Comment (daniel-j-h) [inline src/benchmarks/table.cpp]: What about accepting an optional argument for this number?

**Comment by daniel-j-h** [comment]
Can you rebase on master, clang-format and make this ready for merge into master?

**Comment by daniel-j-h** [comment]
@danpat can you rebase this and merge it in - or close the pr.

At the moment it's just rotting away in the queue.

**Comment by danpat** [comment]
I've ported this to https://github.com/Project-OSRM/osrm-backend/pull/3941 as I'm doing lots of profiling there.

**Pull request closed without merging**